xmtp_mls/src/xmtp_openmls_provider.rs

```diff
@@ -4,28 +4,59 @@ use openmls_traits::OpenMlsProvider;
 use crate::storage::{db_connection::DbConnection, sql_key_store::SqlKeyStore};
 
 #[derive(Debug)]
-pub struct XmtpOpenMlsProvider<'a> {
+pub struct XmtpOpenMlsProvider {
     crypto: RustCrypto,
-    key_store: SqlKeyStore<'a>,
+    key_store: SqlKeyStore<'static>,
 }
 
-impl<'a> XmtpOpenMlsProvider<'a> {
-    pub fn new(conn: &'a DbConnection<'a>) -> Self {
+impl Clone for XmtpOpenMlsProvider {
+    fn clone(&self) -> Self {
+        Self {
+            crypto: RustCrypto::default(),
+            key_store: self.key_store.clone(),
+        }
+    }
+}
+
+impl XmtpOpenMlsProvider {
+    pub fn new(conn: DbConnection) -> Self {
         Self {
             crypto: RustCrypto::default(),
             key_store: SqlKeyStore::new(conn),
         }
     }
 
-    pub(crate) fn conn(&self) -> &DbConnection<'a> {
+    pub(crate) fn conn(&self) -> DbConnection {
         self.key_store.conn()
     }
+
+    pub(crate) fn conn_ref(&self) -> &DbConnection {
+        self.key_store.conn_ref()
+    }
+}
+
+impl OpenMlsProvider for XmtpOpenMlsProvider {
```

Comment:
It does make it easier since the reference and owned version can be used interchangeably, owned & ref variants do need their own impls generally since they can differ in small ways